src/Extension/ExtensionCollection.php

```diff
@@ -0,0 +1,82 @@
+<?php declare(strict_types=1);
+
+namespace OAT\DependencyResolver\Extension;
+
+use OAT\DependencyResolver\Extension\Entity\Extension;
+
+class ExtensionCollection extends \ArrayObject
+{
+    /** @var Extension[] */
+    private $extensions = [];
+
+    /**
+     * @param Extension $extension
+     *
+     * @return $this
+     */
+    public function add(Extension $extension)
+    {
+        $this->offsetSet($extension->getExtensionName(), $extension);
+
+        return $this;
+    }
+
+    /**
+     * @param mixed $index
+     * @param mixed $newval
+     */
+    public function offsetSet($index, $newval)
+    {
+        if (! $newval instanceof Extension) {
+            throw new \TypeError('Extension provided is not an instance of ' . Extension::class . '.');
+        }
+
+        $this->extensions[$index] = $newval;
+    }
+
+    /**
+     * @param string $index
+     *
+     * @return Extension|null
+     */
+    public function offsetGet($index): ?Extension
+    {
+        if (! $this->offsetExists($index)) {
+            return null;
+        }
+
+        return $this->extensions[$index];
+    }
+
+    /**
+     * @param mixed $index
+     *
+     * @return bool
+     */
+    public function offsetExists($index): bool
+    {
+        return array_key_exists($index, $this->extensions);
+    }
+
+    /**
+     * @return \ArrayIterator
+     */
+    public function getIterator(): \ArrayIterator
+    {
+        return new \ArrayIterator($this->extensions);
+    }
+
+    /**
+     * Generates a composer.json contents from the extension collection
+     *
+     * @return false|string
+     */
+    public function generateComposerJson()
```

Comment:
It doesn't really simplifies things, because when invoking json_encode (collection's jsonSerialize returning array_values($this->extensions)), I end up with something like this:
```
[
    {
        "repo1": "branch1"
    },
    {
        "repo2": "branch2"
    }
]
```
instead of expected
```
{
    "repo1": "branch1",
    "repo2": "branch2"
}
which means I'd have to json_encode, json_decode, reduce the array and json_encode again...
Or I'm missing some craft on how to do it properly...

Reply:
I suggest a 50-50 solution, with implementing jsonSerialize on the collection, returning the array `$requires` as created below but not json_encoded in the method. (in my next push, I let you know when).